floss/features/extract.py

```diff
@@ -97,13 +111,58 @@ def extract_function_calls_to(f):
     yield CallsTo(f.vw, [x[0] for x in f.vw.getXrefsTo(f.va, rtype=vivisect.const.REF_CODE)])
 
 
+def extract_function_kinda_tight_loop(f):
+    """
+    Yields tight loop features in the provided function
+    Algorithm by Blaine S.
+    """
+    for bb in f.basic_blocks:
+        # skip first and last BBs
+        if bb.va == f.va or bb.va + bb.size == f.basic_blocks[-1].va:
+            continue
+
+        # skip blocks that don't have exactly 2 successors
+        succs = list(f.get_bb_succs(bb))
```

Comment:
there are a lot of calls to `f.get_bb_succs(...)` in this function. we should cache the results, perhaps explicitly in a local variable here or alternatively wrap the method with a caching decorator as described [here](https://github.com/williballenthin/viv-utils/pull/20#discussion_r734582384)